packages/svelte/src/internal/client/dev/tracing.js

```diff
@@ -0,0 +1,180 @@
+	}
+	const type = (signal.f & DERIVED) !== 0 ? 'derived' : 'state';
+	const current_reaction = /** @type {Reaction} */ (active_reaction);
+	const status =
+		signal.version > current_reaction.version || current_reaction.version === 0 ? 'dirty' : 'clean';
+
+	// eslint-disable-next-line no-console
+	console.groupCollapsed(
+		`%c${type}`,
+		status !== 'clean'
+			? 'color: CornflowerBlue; font-weight: bold'
+			: 'color: grey; font-weight: bold',
+		typeof value === 'object' && STATE_SYMBOL in value ? snapshot(value, true) : value
+	);
+
+	if (type === 'derived') {
+		const deps = new Set(/** @type {Derived} */ (signal).deps);
+		for (const dep of deps) {
+			log_entry(dep);
+		}
+	}
+
+	const read = entry?.read;
+
+	if (read && read.length > 0) {
+		// eslint-disable-next-line no-console
+		console.groupCollapsed('tracked');
+		for (var stack of read) {
+			// eslint-disable-next-line no-console
+			console.log(stack);
+		}
+		// eslint-disable-next-line no-console
+		console.groupEnd();
+	}
+	const created = signal.created;
+
+	if (created) {
+		// eslint-disable-next-line no-console
+		console.groupCollapsed('created');
+		// eslint-disable-next-line no-console
+		console.log(created);
+		// eslint-disable-next-line no-console
+		console.groupEnd();
+	}
+	const updated = signal.updated;
+
+	if (updated) {
+		// eslint-disable-next-line no-console
+		console.groupCollapsed('updated');
+		// eslint-disable-next-line no-console
+		console.log(updated);
+		// eslint-disable-next-line no-console
+		console.groupEnd();
+	}
+	// eslint-disable-next-line no-console
+	console.groupEnd();
+}
+
+/**
+ * @template T
+ * @param {() => T} fn
+ * @param {string} label
+ */
+export function trace(fn, label) {
+	var previously_tracing_expressions = tracing_expressions;
+	try {
+		tracing_expressions = { entries: new Map(), reaction: active_reaction };
+
+		var start = performance.now();
+		var value = fn();
+		var time = (performance.now() - start).toFixed(2);
+
+		if (tracing_expressions.entries.size === 0) {
+			// eslint-disable-next-line no-console
+			console.log(`${label} %cno reactive dependencies (${time}ms)`, 'color: grey');
+		} else {
+			// eslint-disable-next-line no-console
+			console.group(`${label} %c(${time}ms)`, 'color: grey');
+
+			var entries = tracing_expressions.entries;
+
+			tracing_expressions = null;
+
+			for (const [signal, entry] of entries) {
+				log_entry(signal, entry);
+			}
+			// eslint-disable-next-line no-console
+			console.groupEnd();
+		}
+
+		if (previously_tracing_expressions !== null) {
+			for (const [signal, entry] of tracing_expressions.entries) {
+				var prev_entry = previously_tracing_expressions.get(signal);
+
+				if (prev_entry === undefined) {
+					previously_tracing_expressions.set(signal, entry);
+				} else {
+					prev_entry.read.push(...entry.read);
+				}
+			}
+		}
+
+		return value;
+	} finally {
+		tracing_expressions = previously_tracing_expressions;
+	}
+}
+
+export function get_stack() {
```

Comment:
If we pass in a label — `CreatedAt`, `UpdatedAt`, `TracedAt` and use that instead of a generic `TraceInvoked`, then we can get rid of the deep grouping. This...

<img width="460" alt="image" src="https://github.com/user-attachments/assets/0038a2a1-55d6-4f2c-8a3b-01005cb2799c">

...feels better to me than trying to figure out which of these is the one I want to expand, and having to click into it. At the moment it takes a lot of clicks to get anywhere

<img width="485" alt="image" src="https://github.com/user-attachments/assets/da699a25-ccb9-455c-a281-f7bdd550ed81">


```suggestion
/**
 * @param {string} label
 */
export function get_stack(label) {
```

On my local branch I've also reordered stuff. `CreatedAt` comes first because it's a single line, and that helps you quickly identify which piece of state everything else is referring to. `UpdatedAt` comes next because if you're using this rune that's probably the information you're looking for. `TracedAt` comes last because that's generally the least useful piece of information.